pkg/trace/api/telemetry.go

```diff
@@ -91,6 +93,9 @@ func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
 		}
 		req.Header.Set("DD-Agent-Hostname", r.conf.Hostname)
 		req.Header.Set("DD-Agent-Env", r.conf.DefaultEnv)
+		if arn, ok := r.conf.GlobalTags[functionARNKey]; ok {
+			req.Header.Set("DD-Function-ARN", key)
+		}
```

Comment:
This part doesn't compile as the `key` variable isn't present. I believe you meant to use `arn`?

Reply:
```suggestion
		if arn, ok := r.conf.GlobalTags[functionARNKey]; ok {
			req.Header.Set("DD-Function-ARN", arn)
		}
```